_inc/client/plans/accented-card.jsx

```diff
@@ -0,0 +1,17 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+const AccentedCard = props => {
+	const { header, body } = props.children;
```

Comment:
The pattern is called "named slot projection" (e.g. #2 [here](https://medium.com/@martin_hotell/react-children-composition-patterns-with-typescript-56dfc8923c64)). I specified it like this using children instead of props for the semantics. I think of props as state that the component projects into a UI, while children are UI elements for the component to display. 

So making them children instead of props helps to capture the intent of how they are to be used, and if this component needed to display state I would use props instead.

Reply:
And as an aside this is an alternate style that I tried at first:

```
<AccentedCard>
    <AccentedCardHeader>
        { // put header here }
    </AccentedCardHeader>
    <AccentedCardBody>
        { // put body here }
    </AccentedCardBody>
</AccentedCard>
```

I like this better but for it to work well the component needs to check the types of its children, and without TypeScript I decided the named slots projection pattern is cleaner.

Reply:
I still don't understand why the extra complication here. To me using that pattern here doesn't make much sense, mainly because the composition of the components is pretty simple and straightforward, and there's a single use case. Do we need these extra wrapper classes and elements? I think we can totally avoid them.

What you suggested in your follow-up comment is just fine IMHO. It will also prevent from the extra wrapper element.

I'm happy to explore alternatives with you if you have second thoughts about it, but I really feel that the pattern you suggested is just too complicated for the case we have.

Reply:
I really think it breaks the semantics of `props` and `children` to make these `props` since `AccentedCard` is purely a layout component intended to project its `children` and not to render any state. 

`children` in React are handled specially/differently from other `props` to capture these semantics and provide specific syntax for them.

In terms of complexity it is simply the difference between these two uses, and I don't see any extra complexity here, just accurate use of React syntax.

```
<AccentedCard
    header={...}
    body={...}
/>
```


```
<AccentedCard>
{ {
    header: ...,
    body: ...,
} }
</AccentedCard>
```

I know this is just semantics but I think these semantics are important. `props` and `children` aren't the same thing even if they can technically be used interchangeably.

Reply:
It is just not a familiar pattern around here and it's really unnecessary in this simple case. Passing the props directly makes more sense to me. But if you ask me, the need for this accented card component is unclear - we could just use the wrapper around the components directly, and avoid the need for an extra component. This will greatly simplify everything. So you would just render your header and body component, wrapped by a div with some class. Let me know if that makes sense.